### PR TITLE
Add product for Kafka Schema Registry and Avro format

### PR DESCRIPTION
**Pull request by kokosing**
Add product for Kafka Schema Registry and Avro format

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java [hunk 1]
```diff
@@ -13,162 +13,259 @@
...
-        QueryResult queryResult = query(format("SELECT a[1], a[2], m['key1'] FROM (SELECT c_array as a, c_map as m FROM %s.%s) t", KAFKA_CATALOG, STRUCTURAL_AVRO_TABLE_NAME));
-        assertThat(queryResult).containsOnly(row(100, 102, "value1"));
+        @Override
+        public byte[] serialize(String topic, Schema schema, Map<String, Object> values)
+                throws IOException
+        {
+            try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+                out.write((byte) 0);
+                SchemaId schemaId = registerSchema(topic, schema);
+                out.write(ByteBuffer.allocate(4).putInt(schemaId.getId()).array());
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java:231]:
```suggestion
                out.write(Ints.toByteArray(schemaId.getId()));
```

File: testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java [hunk 2]
```diff
@@ -13,162 +13,259 @@
...
+
+                DirectBinaryEncoder encoder = createDirectBinaryEncoder(out);
```
Comment (Praveen2112) [inline testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java:233]:
```suggestion
                DirectBinaryEncoder encoder = EncoderFactory.get().directBinaryEncoder(out, null);
```

**Review by Praveen2112 — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -1556,6 +1556,12 @@
                 <artifactId>snakeyaml</artifactId>
                 <version>1.26</version>
             </dependency>
+
```
Comment (skrzypo987) [inline pom.xml:1559]: `Add product for` -> `Add product tests for` in commit message

**Review by skrzypo987 — APPROVED** [review]

**Comment by kokosing** [comment]
Flaky https://github.com/trinodb/trino/issues/6412

**Pull request merged**